### ISSUE TITLE
Client/Server Architecture

### ISSUE THREAD

**Issue by brad999** [brad999/nikita-client#29]
- Decouple "passive" modules to run on a "server" version of jasper and the "active" modules to run on the client version. This will allow for global passive actions (email notification, offline notification, etc) to be set at a global level for clients. Server will update clients when there is a passive notification that applies to them. This will also allow clients to message each other using the server and the notification placed in their queue.
- Clients will report to the server when they come online so that the server can have an inventory of the clients online. Clients should be inventoried with a common name (i.e. bedroom, living room, etc) as well as their static IP.

**Comment by asantaga** [comment]
Can you run the client & server on the same PI? i think that would be "desirable" that way cloner's could run it on one pi if they dont need the C/S functionality...

**Comment by brad999** [comment]
Yes, that is exactly the plan. I will even be running it this way for a while until I get another mic for my other Pi. Then I will move the server program to my actual home server.
